packages/firestore/tools/console.build.js

```diff
@@ -20,40 +20,51 @@
  * This file creates a build target for it.
  */
 const rollup = require('rollup');
-const typescriptPlugin = require('rollup-plugin-typescript2');
-const typescript = require('typescript');
-const json = require('rollup-plugin-json');
-const alias = require('@rollup/plugin-alias');
-const resolve = require('rollup-plugin-node-resolve');
 const { uglify } = require('rollup-plugin-uglify');
+const resolve = require('rollup-plugin-node-resolve');
 const fs = require('fs');
 const util = require('util');
 const fs_writeFile = util.promisify(fs.writeFile);
 
 const rollupUtil = require('../rollup.shared');
 
-const plugins = [
-  alias(rollupUtil.generateAliasConfig('browser')),
-  resolve(),
-  typescriptPlugin({
-    typescript
-  }),
-  json(),
-  uglify({
-    output: {
-      ascii_only: true // escape unicode chars
-    }
-  })
-];
-
 const EXPORTNAME = '__firestore_exports__';
 
-const inputOptions = {
+const esm2017OutputFile = 'dist/standalone.esm2017.js';
+const esm5OutputFile = 'dist/standalone.js';
+
+const es2017InputOptions = {
   input: 'index.console.ts',
-  plugins
+  plugins: rollupUtil.es2017Plugins('browser', /* mangled= */ true),
+  external: rollupUtil.resolveBrowserExterns,
+  treeshake: {
+    moduleSideEffects: false
+  }
+};
+
+const es2017OutputOptions = {
+  file: esm2017OutputFile,
+  format: 'es'
+};
+
+const es2017toEs5InputOptions = {
+  input: esm2017OutputFile,
+  plugins: [
+    resolve(),
+    ...rollupUtil.es2017ToEs5Plugins(/* mangled= */ true),
+    uglify({
+      output: {
+        ascii_only: true // escape unicode chars
```

Comment:
nice! Found the solution for https://github.com/firebase/firebase-js-sdk/issues/414